### ISSUE TITLE
Readonly on definition class level is lost with some configurations

### ISSUE THREAD

**Issue by fengzhou-msft** [Azure/autorest#4269]
**Before filling a bug**

- [ ] have you checked the [faq](https://github.com/Azure/autorest/tree/main/docs/faq.md) for known issues.
- [ ] have you checked existing issues

**Describe the bug**
When working with generating the [common types](https://raw.githubusercontent.com/Azure/azure-rest-api-specs/ac3be41ee22ada179ab7b970e98f1289188b3bae/specification/common-types/resource-management/v2/types.json) in .NET Mgmt TRack2 SDK, we found that with the following directive, the properties of the generated SystemData model is no longer readonly. It will have get and set methods for each property while without the configuration, each property only has get methods.
```
  - from: types.json
    where: $.definitions.systemData
    transform: >
      $["x-csharp-usage"] = "model,input,output"
```

If we apply the readonly attribute to each property with the following directive, then readonly attrribute on each property is not lost with the above configuration.
```
  - from: types.json
    where: $.definitions.systemData.properties.*
    transform: >
      $["readOnly"] = true
```

The autorest.md file used:
https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/resourcemanager/Azure.ResourceManager/src/autorest.md

I'm also wondering what is the difference of readonly on the definition class level and readonly on each property?

**Expected behavior**
The proterties of SystemData are readonly with or whithout the x-csharp-usage configuration when systemData is defined as readonly.

**Additional context**

**Comment by timotheeguerin** [comment]
Do you mean the properties were readonly by default before? if so that must be a change in the csharp generator.

For readonly on the properties vs schema:

From the openapi specification:

readOnly | boolean | Relevant only for Schema "properties" definitions. Declares the property as "read only". This means that it MAY be sent as part of a response but SHOULD NOT be sent as part of the request. If the property is marked as readOnly being true and is in the required list, the required will take effect on the response only. A property MUST NOT be marked as both readOnly and writeOnly being true. Default value is false.
-- | -- | --

Basically just means if schema is readonly it shouldn't be used to send request but if only some properties are readonly then those props should be omitted from the request

**Comment by fengzhou-msft** [comment]
@timotheeguerin thanks for the explanation. It seems the issue is caused by the behavior change in autorest.csharp when interpreting `x-csharp-usage`.
